### ISSUE TITLE
Fix S2857 FP: @parameter with nameof and FN after [tablename]

### ISSUE THREAD

**Issue by Dzeneralen** [SonarSource/sonar-dotnet#6126]
### Description

**Rule: S2857**

Bug with false positive and negative for SQL when adding `@` before nameof. The use of `@` is for Dapper SQL statements.

### Repro steps

```csharp
class Example
{

    class Model
    {
        public int ColumnName { get; set; }
        public int OtherColumn { get; set; }
    }

    public void Case()
    {
        using var connection = new SqlConnection("...");

        const string sql = "UPDATE [some_table]" +   // FN
            $"SET [some_column] = @{nameof(Model.ColumnName)}," +
            $" [other_column] = @{nameof(Model.OtherColumn)}";  // FP due to @{nameof()}

        connection.Execute(sql, new Model());
    }
}
```

### Expected behavior

- Line with `UPDATE` is flagged.
- Line with `[other_column]` is not flagged 

### Actual behavior

- Does not flag the `UPDATE` line which should be flagged (False negative).
- Flags the line below with `[other_column]` even though it appears to be valid SQL and following its rule. (S2857: Add a space before 'OtherColumn'

### Known workarounds

None

### Related information

* SonarLint for Visual Studio 2022, version 6.8.0.53188
* Visual Studio 2022
* dotnet 6
* Windows 10

**Comment by andrei-epure-sonarsource** [comment]
Thank you for opening this issue, @Dzeneralen . Your feedback helps us improve our products. 

We will look at it as soon as possible. After we confirm the issue, we will add labels and put it in our backlog.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Dzeneralen,

Thank you for reporting this. I confirm it as FP and FN.

The FP should not be raised.
The FN should be raised the online with `UPDATE ` statement.
I've updated the original message

**Comment by SawNawDfE** [comment]
Hi, we're still getting this false positive here: https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&pullRequest=149&id=SkillsFundingAgency_das-provide-feedback-employer&open=AYZQplAEFUASU11F5jyQ

Please do you have a timeline for releasing this fix?

**Comment by pavel-mikula-sonarsource** [comment]
We don't have any specific ETA, unfortunatelly.

**Comment by PiotrKula-ECO** [comment]
Hello we also still getting this issue. Using sonar cloud latest version. 
![image](https://github.com/SonarSource/sonar-dotnet/assets/123736819/90484e6c-f4ef-4dcb-b211-38923964e820)
